branchpro/results/combined-inference.ipynb

```diff
@@ -159,6 +159,86 @@
+    "proportion_time_r_more_than_1_UpperCI_ON = []\n",
+    "\n",
+    "large_epsilon_range = np.arange(0.1, 2.5, 0.1)\n",
+    "\n",
+    "for epsilon in large_epsilon_range:\n",
+    "    inference = branchpro.LocImpBranchProPosteriorMultSI(\n",
+    "        inc_data=inc_data,\n",
+    "        imported_inc_data=imported_inc_data,\n",
+    "        epsilon=epsilon,\n",
+    "        daily_serial_intervals=serial_intervals,\n",
+    "        alpha=a,\n",
+    "        beta=b)\n",
+    "\n",
+    "    inference.run_inference(tau=tau)\n",
+    "    intervals = inference.get_intervals(central_prob=0)\n",
+    "    intervals = intervals.append(inference.get_intervals(central_prob=.25))\n",
+    "    intervals = intervals.append(inference.get_intervals(central_prob=.95))\n",
+    "    temp = inference.get_intervals(central_prob=.95)\n",
+    "\n",
+    "    intervals['Epsilon'] = [epsilon] * len(intervals.index)\n",
+    "\n",
+    "    large_all_intervals = large_all_intervals.append(intervals)\n",
+    "\n",
+    "    # Last time R > 1\n",
+    "    if len(large_all_intervals.loc[(large_all_intervals['Epsilon']==epsilon) & (large_all_intervals['Mean']>1) & (large_all_intervals['Central Probability']==0.95)]['Time Points'].tolist())==0:\n",
+    "        last_time_r_more_than_1_ON.append(None)\n",
+    "    else:\n",
+    "        last_time_r_more_than_1_ON.append(large_all_intervals.loc[(large_all_intervals['Epsilon']==epsilon) & (large_all_intervals['Mean']>1) & (large_all_intervals['Central Probability']==0.95)]['Time Points'].tolist()[-1]+1)\n",
+    "\n",
+    "    if len(large_all_intervals.loc[(large_all_intervals['Epsilon']==epsilon) & (large_all_intervals['Lower bound CI']>1) & (large_all_intervals['Central Probability']==0.95)]['Time Points'].tolist())==0:\n",
+    "        last_time_r_more_than_1_LowerCI_ON.append(None)\n",
+    "    else:\n",
+    "        last_time_r_more_than_1_LowerCI_ON.append(large_all_intervals.loc[(large_all_intervals['Epsilon']==epsilon) & (large_all_intervals['Lower bound CI']>1) & (large_all_intervals['Central Probability']==0.95)]['Time Points'].tolist()[-1]+1)\n",
+    "\n",
+    "    if len(large_all_intervals.loc[(large_all_intervals['Epsilon']==epsilon) & (large_all_intervals['Upper bound CI']>1) & (large_all_intervals['Central Probability']==0.95)]['Time Points'].tolist())==0:\n",
+    "        last_time_r_more_than_1_UpperCI_ON.append(None)\n",
+    "    else:\n",
+    "        last_time_r_more_than_1_UpperCI_ON.append(large_all_intervals.loc[(large_all_intervals['Epsilon']==epsilon) & (large_all_intervals['Upper bound CI']>1) & (large_all_intervals['Central Probability']==0.95)]['Time Points'].tolist()[-1]+1)\n",
+    "\n",
+    "\n",
+    "    # Last time R < 1\n",
+    "    if len(large_all_intervals.loc[(large_all_intervals['Epsilon']==epsilon) & (large_all_intervals['Mean']<1) & (large_all_intervals['Central Probability']==0.95)]['Time Points'].tolist())==0:\n",
```

Comment:
I agree with this point and it shouldn't be too complicated. Would I need to rerun the notebook though afterwards?

Reply:
You'd need to update the notebook to use the new methods, and make sure it doesn't have any errors when you start running the cell. Waiting for it to finish shouldn't be necessary since the behavior will be the same.